### PR TITLE
Modular JAR: Add module-info module descriptor and require Java 9+

### LINKED ISSUE #125: Add jlink-compatible Java9/Jigsaw module-info (not automatic modules)
Latest version has automatic modules and can be used in Java 9 projects (#114). However, this is not compatible with the jlink tool.

__Steps to reproduce the problem (provide example input):__

- create a Java 9 based project that uses modules
- Use jlink

__Expected behavior:__

jlink links the prokect

__Actual behavior:__

jlink reports an error:
```
> Task :link FAILED
Error: automatic module cannot be used with jlink: org.commonmark.ext.gfm.strikethrough from file:///Users/axel/.gradle/caches/modules-2/files-2.1/com.atlassian.commonmark/commonmark-ext-gfm-strikethrough/0.11.0/2b3f085711511202c380f2540760d64302c6bb7e/commonmark-ext-gfm-strikethrough-0.11.0.jar

```

### PR DESCRIPTION
**Pull request by robinst**
We've waited long enough for the ecosystem and tools such as Android to catch up. Making it proper modular means people can run jlink on it. Fixes #125.

### CODE REVIEW

**Comment by ice1000** [comment]
Nice!! Oh yeah!!

**Comment by ice1000** [comment]
Thank you very much!!! You're the real hero!

**Comment by boyshell** [comment]
您好！邮件已收到。

File: commonmark-android-test/app/build.gradle
Comment (robinst) [inline commonmark-android-test/app/build.gradle:1]: @noties could you have a look at these and other changes in the Android directory? [The build](https://github.com/commonmark/commonmark-java/actions/runs/8121393434/job/22199696869) is green with these changes, but I'm not sure I did everything properly.

There was one problem of using `String#repeat` (introduced in Java 11), which is not yet available on the minimum API level, which the Android build detected. I changed the code not to use that for now, and now the build is green.

**Pull request merged**